### PR TITLE
test(driver): refactor rank driver tests

### PR DESCRIPTION
**Pull request by Yongxuanzhang**
- Use `Document` instead of jina_pb2.DocumentProto()
- Refactor code to make it clean

### CODE REVIEW

File: tests/unit/drivers/rank/aggregate/test_aggregate_matches_rank_driver.py [hunk 1]
```diff
@@ -67,38 +67,22 @@ def create_document_to_score_same_depth_level():
...
-    match5.score.value = 10
-
-    return Document(doc)
+    for match_id, parent_id, match_score, match_length in [
+        (2, 20, 30, 3),
+        (3, 20, 40, 4),
+        (4, 30, 20, 2),
+        (5, 30, 10, 1),
+    ]:
+        with Document() as match:
```
Comment (Yongxuanzhang) [inline tests/unit/drivers/rank/aggregate/test_aggregate_matches_rank_driver.py:79]:
```suggestion
        match = Document()
```

**Review by Yongxuanzhang — COMMENTED** [review]
TODO

File: tests/unit/drivers/rank/aggregate/test_aggregate_matches_rank_driver.py [hunk 2]
```diff
@@ -67,38 +67,22 @@ def create_document_to_score_same_depth_level():
...
+            match.id = str(match_id) * 16
```
[the remark this answers is not in the document]
Comment (Yongxuanzhang) [reply tests/unit/drivers/rank/aggregate/test_aggregate_matches_rank_driver.py]: Done

File: tests/unit/drivers/rank/aggregate/test_aggregate_matches_rank_driver.py [hunk 3]
```diff
@@ -112,7 +96,7 @@ def test_collect_matches2doc_ranker_driver_mock_ranker():
     assert dm[0].id == '20' * 8
```
[the remark this answers is not in the document]
Comment (Yongxuanzhang) [reply tests/unit/drivers/rank/aggregate/test_aggregate_matches_rank_driver.py]: Done

File: tests/unit/drivers/rank/aggregate/test_aggregate_matches_rank_driver.py (hunk 1 above)
Comment (Yongxuanzhang) [reply tests/unit/drivers/rank/aggregate/test_aggregate_matches_rank_driver.py]: Done

File: tests/unit/drivers/rank/aggregate/test_aggregate_matches_rank_driver.py [hunk 4]
```diff
@@ -67,38 +67,22 @@ def create_document_to_score_same_depth_level():
...
+        (4, 30, 20, 2),
+        (5, 30, 10, 1),
+    ]:
+        match = Document()
+        match.id = match_id
+        match.parent_id = parent_id
+        match.length = match_length
+        match.score = NamedScore(value=match_score, ref_id=doc.id)
+        doc.matches.append(match)
+    return doc
```
Comment (bwanglzu) [inline tests/unit/drivers/rank/aggregate/test_aggregate_matches_rank_driver.py:85]: parameterize this one

**Review by bwanglzu — CHANGES_REQUESTED** [review]

File: tests/unit/drivers/rank/aggregate/test_aggregate_matches_rank_driver.py (hunk 4 above)
Comment (Yongxuanzhang) [reply tests/unit/drivers/rank/aggregate/test_aggregate_matches_rank_driver.py:85]: Sure!

Comment (Yongxuanzhang) [reply tests/unit/drivers/rank/aggregate/test_aggregate_matches_rank_driver.py:85]: I will fix it tomorrow

**Review by florian-hoenicke — APPROVED** [review]

File: tests/unit/drivers/rank/test_matches2doc_rank_drivers.py [hunk 5]
```diff
@@ -62,14 +63,14 @@ def test_chunk2doc_ranker_driver_mock_exec():
...
+    assert doc.matches[0].id == '2' * 16
+    assert doc.matches[0].score.value == -4
+    assert doc.matches[1].id == '3' * 24
+    assert doc.matches[1].score.value == -4
     assert doc.matches[2].id == '5' * 16
-    assert doc.matches[2].score.value == -3
+    assert doc.matches[2].score.value == -4
     assert doc.matches[3].id == '4' * 8
-    assert doc.matches[3].score.value == -4
+    assert doc.matches[3].score.value == -12
```
Comment (cristianmtr) [inline tests/unit/drivers/rank/test_matches2doc_rank_drivers.py:73]: Why did the value change?

Comment (Yongxuanzhang) [reply tests/unit/drivers/rank/test_matches2doc_rank_drivers.py:73]: Good question! 
This test uses `executor = MockAbsoluteLengthRanker()` to calculate `distance`, and previously we set `match.length = match_score`, and I changed it to `match.length = match_length`. Meanwhile the doc length is also changed `doc.length = 20`.
Which one do you think makes more sense?

File: tests/unit/drivers/rank/aggregate/test_aggregate_matches_rank_driver.py (hunk 4 above)
Comment (Yongxuanzhang) [reply tests/unit/drivers/rank/aggregate/test_aggregate_matches_rank_driver.py:85]: You mean @pytest.fixture this function?

File: tests/unit/drivers/rank/test_matches2doc_rank_drivers.py (hunk 5 above)
Comment (JoanFM) [reply tests/unit/drivers/rank/test_matches2doc_rank_drivers.py:73]: Let's first merge the code without the value changes. Then we can do a PR changing how the test itself is designed (which I agree has a huge improvement capacity)

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: tests/unit/drivers/rank/test_matches2doc_rank_drivers.py (hunk 5 above)
Comment (Yongxuanzhang) [reply tests/unit/drivers/rank/test_matches2doc_rank_drivers.py:73]: Sure! That makes sense, I will restore these changes. @JoanFM

File: tests/unit/drivers/rank/aggregate/test_aggregate_matches_rank_driver.py (hunk 4 above)
Comment (bwanglzu) [reply tests/unit/drivers/rank/aggregate/test_aggregate_matches_rank_driver.py:85]: @Yongxuanzhang sorry I didn't see it clearly, yes you can create a fixture, or leave it like this, looks good.

Comment (Yongxuanzhang) [reply tests/unit/drivers/rank/aggregate/test_aggregate_matches_rank_driver.py:85]: Ok Thanks!

File: tests/unit/drivers/rank/test_matches2doc_rank_drivers.py [hunk 6]
```diff
@@ -61,7 +62,7 @@ def create_document_to_score():
         with Document() as match:
             match.id = str(match_id) * match_length
             match.length = match_score
```
Comment (Yongxuanzhang) [inline tests/unit/drivers/rank/test_matches2doc_rank_drivers.py:64]: Maybe we can revisit this later.
`match.length = match_score`

File: tests/unit/drivers/rank/test_matches2doc_rank_drivers.py (hunk 5 above)
Comment (Yongxuanzhang) [reply tests/unit/drivers/rank/test_matches2doc_rank_drivers.py]: Just reverted these values change in the latest commit. 😄

File: tests/unit/drivers/rank/aggregate/test_aggregate_matches_rank_driver.py [hunk 7]
```diff
@@ -109,10 +93,10 @@ def test_collect_matches2doc_ranker_driver_mock_ranker():
     driver()
     dm = list(doc.matches)
     assert len(dm) == 2
-    assert dm[0].id == '20' * 8
+    assert dm[0].id == '20'
     assert dm[0].score.value == 3
-    assert dm[1].id == '30' * 8
-    assert dm[1].score.value == 1
+    assert dm[1].id == '30'
+    assert dm[1].score.value == 2
```
Comment (JoanFM) [inline tests/unit/drivers/rank/aggregate/test_aggregate_matches_rank_driver.py:99]: why does this change?

File: tests/unit/drivers/rank/aggregate/test_aggregate_matches_rank_driver.py [hunk 8]
```diff
@@ -67,38 +67,22 @@ def create_document_to_score_same_depth_level():
...
-
-    match4 = doc.matches.add()
-    match4.id = str(4) * 16
-    match4.parent_id = str(30) * 8
-    match4.length = 2
-    match4.score.ref_id = doc.id
-    match4.score.value = 20
-
-    match5 = doc.matches.add()
-    match5.id = str(4) * 16
```
Comment (Yongxuanzhang) [inline tests/unit/drivers/rank/aggregate/test_aggregate_matches_rank_driver.py:95]: This line is not correct in the previous test. @JoanFM

File: tests/unit/drivers/rank/aggregate/test_aggregate_matches_rank_driver.py (hunk 7 above)
Comment (Yongxuanzhang) [reply tests/unit/drivers/rank/aggregate/test_aggregate_matches_rank_driver.py:99]: The id of  match5 is set to the same as match4 in the previous code`match5.id = str(4) * 16`

**Review by JoanFM — APPROVED** [review]

**Review by bwanglzu — APPROVED** [review]
LGTM!

**Pull request merged**